ui_components/src/ramen/RamenShop.js

```diff
@@ -0,0 +1,231 @@
+import { ModalProvider, useModal } from "../utils/modalContext.js";
+import { apiFetch } from "../utils/network.js";
+import { ResourceBar } from "../utils/resourceBar.js";
+import { parseKeywords } from "../utils/parseKeywords.js";
+
+function RamenShopReactContainer({
+    ramenShopAPI,
+    playerData,
+    playerResourcesData,
+    characterRamenData,
+    ramenOwnerDetails,
+    basicRamenOptions,
+    specialRamenOptions,
+    mysteryRamenDetails
+}) {
+    return (
+        <ModalProvider>
+            <RamenShop
+                ramenShopAPI={ramenShopAPI}
+                playerData={playerData}
+                playerResourcesData={playerResourcesData}
+                characterRamenData={characterRamenData}
+                ramenOwnerDetails={ramenOwnerDetails}
+                basicMenuOptions={basicRamenOptions}
+                specialMenuOptions={specialRamenOptions}
+                mysteryRamenDetails={mysteryRamenDetails}
+            />
+        </ModalProvider>
+    );
+}
+
```

Comment:
Not going to block this PR, but for future PRs please start adding Flow types for all your React components, makes them a lot easier to work with and know what's in the props